NumLib/ODESolver/MatrixTranslator.h

```diff
@@ -0,0 +1,435 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef NUMLIB_MATRIXTRANSLATOR_H
+#define NUMLIB_MATRIXTRANSLATOR_H
+
+#include <memory>
+
+#include "Types.h"
+#include "TimeDiscretization.h"
+
+#include "MathLib/LinAlg/BLAS.h"
+
+namespace NumLib
+{
+
+//! \addtogroup ODESolver
+//! @{
+
+
+/*! Translates matrices assembled by a provided ODE (or other equation)
+ *  to some other matrices suitable to be passed on to nonlinear solvers.
+ *
+ * \tparam Matrix the type of matrices occuring in the linearization of the equation.
+ * \tparam Vector the type of the solution vector of the ODEequation
+ * \tparam ODETag a tag indicating the type of equation.
+ */
+template<typename Matrix, typename Vector, ODESystemTag ODETag>
+class MatrixTranslator;
```

Comment:
I'll keep the old name, because it's less work ^^ , and the typenames, hence the lines of code, are already long enough (e.g. `MatrixTranslatorCrankNicolson`).